app/daemon_test.go

```diff
@@ -739,94 +739,120 @@ func TestTimeToSweepSessions(t *testing.T) {
 func TestDBusEventLoop(t *testing.T) {
 	currentUser, err := user.Current()
 	if err != nil {
-		t.Errorf("cant get current user: %s", err.Error())
-		return
+		t.Fatalf("cant get current user: %s", err)
 	}
 
 	testCases := []struct {
 		name    string
-		err     error
 		token   string
+		err     error
 		timeout time.Duration
 	}{
 		{
-			name:  "stopped-gracefully",
-			token: "the-token",
+			name:    "stopped-gracefully",
+			token:   "the-token",
+			timeout: 4 * time.Second,
 		},
 		{
 			name:    "token_not_returned_wait_forever",
 			token:   "",
 			timeout: 15 * time.Second,
 		},
+		{
+			name:    "token_not_returned_try_reconnection",
+			token:   "",
+			timeout: 2 * time.Second,
+		},
 	}
 
 	for _, tc := range testCases {
-		if tc.name == "token_not_returned_wait_forever" {
-			timeout := time.After(tc.timeout)
-			done := make(chan bool)
-			go func() {
-				t.Run(tc.name, func(t *testing.T) {
-					d := NewDaemon(&config.MenderShellConfig{
-						MenderShellConfigFromFile: config.MenderShellConfigFromFile{
-							ShellCommand: "/bin/sh",
-							User:         currentUser.Name,
-							Terminal: config.TerminalConfig{
-								Width:  24,
-								Height: 80,
+		t.Run(tc.name, func(t *testing.T) {
+			d := NewDaemon(&config.MenderShellConfig{
+				MenderShellConfigFromFile: config.MenderShellConfigFromFile{
+					ShellCommand: "/bin/sh",
+					User:         currentUser.Name,
+					Terminal: config.TerminalConfig{
+						Width:  24,
+						Height: 80,
+					},
+				},
+			})
+
+			dbusAPI := &dbusmocks.DBusAPI{}
+			defer dbusAPI.AssertExpectations(t)
+			client := &authmocks.AuthClient{}
+
+			switch tc.name {
+			case "token_not_returned_wait_forever":
+				timeout := time.After(tc.timeout)
+				done := make(chan bool)
+				go func() {
+					t.Run(tc.name, func(t *testing.T) {
```

Comment:
I don't get why we have `t.Run(...)` inside another `t.Run(...)`, can you explain me the intention?

Reply:
Oh it really isn't needed, so changed it now. Good catch, thanks :)